src/Products/Traits/With_Json_Data_Sets.php

```diff
@@ -0,0 +1,46 @@
+<?php
+/**
+ * Provides methods to load large data sets from JSON files.
+ *
+ * The JSON files are expected to be in: `<test_case_dir>/data-sets/<test_case_class>/*.json`.
+ *
+ * @package Tribe\Test\Products\Traits
+ */
+
+namespace Tribe\Test\Products\Traits;
+
+/**
+ * Trait With_Json_Data_Sets
+ *
+ * @package Tribe\Test\Products\Traits
+ */
+trait With_Json_Data_Sets {
+
+	/**
+	 * Yields the JSON data sets stored in the test case case `data-sets/<short_class_name>` sibling directory.
+	 *
+	 * @param string $test_method The test case method, usually passed by PHPUnit, if specified and if a data-sets
+	 *                            sub-dir dedicated to the case exists, then data-sets will be loaded from that.
+	 *
+	 * @return \Generator The decoded JSON data sets.
+	 *
+	 * @throws \RuntimeException If the data set directory does not exist and cannot be created.
+	 */
+	public function json_data_sets( $test_method = null ) {
+		$r   = new \ReflectionClass( $this );
+		$dir = implode( '/', [ dirname( $r->getFileName() ), 'data-sets', $r->getShortName() ] );
+
+		if ( ! ( is_dir( $dir ) || mkdir( $dir, 0777, true ) ) ) {
+			throw new \RuntimeException( "Cannot create data-set dir: {$dir}" );
```

Comment:
Not sure why we're creating the folder here?
There's nothing in the docblock about that, and this function doesn't generate data, just reads/formats it, correct?

Reply:
This function only reads the data, correct.

Yet the expected location of files is not immediate from the code and I just wanted to lower the debug barrier ("Where should I store the JSON data sets to have them read here?!") by doing it.
It's pretty common to scaffold and read to speed up test development.

Reply:
I'm adding a note about this in the code block.